app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckClaimantDetailsController.scala

```diff
@@ -173,49 +208,94 @@ object CheckClaimantDetailsController {
             Some(declaration.displayResponseDetail.declarantDetails.establishmentAddress)
         }
       }
-      .flatten
+      .getOrElse(None)
   }
 
   def extractContactDetails(fillingOutClaim: FillingOutClaim): NamePhoneEmail = {
     val draftC285Claim = fillingOutClaim.draftClaim.fold(identity)
     val email          = fillingOutClaim.signedInUserDetails.verifiedEmail
-    Applicative[Option]
-      .map2(draftC285Claim.displayDeclaration, draftC285Claim.declarantTypeAnswer) { (declaration, declarantType) =>
-        declarantType match {
-          case DeclarantTypeAnswer.Importer | DeclarantTypeAnswer.AssociatedWithImporterCompany =>
-            val consignee = declaration.displayResponseDetail.consigneeDetails
-            val name      = consignee.flatMap(_.contactDetails).flatMap(_.contactName)
-            val phone     = consignee.flatMap(_.contactDetails).flatMap(_.telephone)
-            NamePhoneEmail(name, phone.map(PhoneNumber(_)), Some(email))
-          case DeclarantTypeAnswer.AssociatedWithRepresentativeCompany                          =>
-            val declarant = declaration.displayResponseDetail.declarantDetails
-            val name      = declarant.contactDetails.flatMap(_.contactName)
-            val phone     = declarant.contactDetails.flatMap(_.telephone)
-            NamePhoneEmail(name, phone.map(PhoneNumber(_)), Some(email))
-        }
+    draftC285Claim.mrnContactDetailsAnswer
+      .map { contact =>
+        NamePhoneEmail(Option(contact.fullName), Option(contact.phoneNumber), Option(contact.emailAddress))
       }
+      .orElse(
+        Applicative[Option]
+          .map2(draftC285Claim.displayDeclaration, draftC285Claim.declarantTypeAnswer) { (declaration, declarantType) =>
```

Comment:
What would be result of map2 if one field is None? Would it be None?

Reply:
Yes